src/cbmc/cbmc_solvers.cpp

```diff
@@ -189,8 +187,6 @@ class cbmc_solver_with_filet:public cbmc_solverst::solvert
   ~cbmc_solver_with_filet()
   {
     // delete the prop before the file
-    delete prop_conv_ptr;
```

Comment:
The problem is that it is not only prop. Some solvers need prop+aig, others a file,...

The simplest fix for the concrete case would be to turn cbmc_solvers.h:62 into
```
if(prop_conv_ptr!=NULL)
  delete prop_conv_ptr;
```

More generally, solvert could provide a facility to store any number of pointers in a given order and it is then responsible to clean them up in reverse order.